tests/std/tests/P0896R4_common_iterator/test.cpp

```diff
@@ -4,65 +4,109 @@
+            using OCit      = common_iterator<ConstIter, OSen>;
+            P input[3]      = {{0, 1}, {0, 2}, {0, 3}};
 
             // [common.iter.types]
-            using iconcept = typename iterator_traits<Cit>::iterator_concept;
-            if constexpr (forward_iterator<Iter>) {
-                STATIC_ASSERT(same_as<iconcept, forward_iterator_tag>);
-            } else {
-                STATIC_ASSERT(same_as<typename iterator_traits<Cit>::iterator_concept, input_iterator_tag>);
-            }
+            {
+                using iconcept = typename iterator_traits<Cit>::iterator_concept;
+                if constexpr (forward_iterator<Iter>) {
+                    STATIC_ASSERT(same_as<iconcept, forward_iterator_tag>);
+                } else {
+                    STATIC_ASSERT(same_as<typename iterator_traits<Cit>::iterator_concept, input_iterator_tag>);
+                }
 
-            using icat = typename iterator_traits<Cit>::iterator_category;
-            if constexpr (derived_from<icat, forward_iterator_tag>) {
-                STATIC_ASSERT(same_as<icat, forward_iterator_tag>);
-            } else {
-                STATIC_ASSERT(same_as<icat, input_iterator_tag>);
-            }
+                using icat = typename iterator_traits<Cit>::iterator_category;
+                if constexpr (derived_from<icat, forward_iterator_tag>) {
+                    STATIC_ASSERT(same_as<icat, forward_iterator_tag>);
+                } else {
+                    STATIC_ASSERT(same_as<icat, input_iterator_tag>);
+                }
 
-            using ipointer = typename iterator_traits<Cit>::pointer;
-            if constexpr (_Has_op_arrow<Iter>) {
-                STATIC_ASSERT(same_as<ipointer, decltype(declval<const Iter&>().operator->())>);
-            } else {
-                STATIC_ASSERT(same_as<ipointer, void>);
+                using ipointer = typename iterator_traits<Cit>::pointer;
+                if constexpr (_Has_op_arrow<Iter>) {
+                    STATIC_ASSERT(same_as<ipointer, decltype(declval<const Iter&>().operator->())>);
+                } else {
+                    STATIC_ASSERT(same_as<ipointer, void>);
+                }
             }
 
-            { // [counted.iter.const]
-                [[maybe_unused]] Cit defaultConstructed{};
+            { // [common.iter.const]
+                Cit defaultConstructed{};
+                Cit iterConstructed{Iter{input}};
+                Cit sentinelConstructed(Sen{});
+                defaultConstructed = iterConstructed;
+
+                OCit conversionConstructed{defaultConstructed};
+                conversionConstructed = iterConstructed;
 
-                [[maybe_unused]] Cit iterConstructed{Iter{input}};
-                [[maybe_unused]] Cit sentinelConstructed(Sen{});
+                OCit conversionConstructedSentinel{sentinelConstructed};
+                conversionConstructed = iterConstructed;
             }

```

Comment:
I think we're still missing plain-vanilla non-converting copy constructor coverage here.